### PR TITLE
SQLite3 task type implemented

### PR DESCRIPTION
**Pull request by kumare3**
Plugin that allows queries to be run on SQLite3 database. This can be used to query data from MySQL, PostGres etc as well

### CODE REVIEW

File: flytekit/taskplugins/sqlite/task.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    If the Mode is marked dynamic, then a variable called ``sqlite: FlyteFile[".sqlite"]`` will be automatically added
+    to the input interface of the task. Failing to supply this at runtime will raise an Assertion
+
+        Args:
+            sqlite_db_mode: STATIC should be use for constant locations of file, dynamic to allow dynamically provided
+                            sqlite3 file locations. The file will have to comply with FlyteFile[.db|.zip|.tar.gz|.bz2]
+                            format
+            static_file_path: default FlyteFile that will be downloaded on execute
+            compressed: Boolean that indicates if the given file is a compressed archive
```
Comment (katrogan) [inline flytekit/taskplugins/sqlite/task.py:34]: worth including which compressed file types are supported? or maybe just link to shutil docs?

File: flytekit/taskplugins/sqlite/task.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+            if self._SQLITE_INPUT_FILE in inputs:
+                raise ValueError(f"Input name `{self._SQLITE_INPUT_FILE}` is reserved as an auto-injected input")
+            inputs[self._SQLITE_INPUT_FILE] = FlyteFile
+        if output_schema_type is None:
+            output_schema_type = FlyteSchema
+        outputs = kwtypes(results=output_schema_type)
+        if metadata is None:
+            metadata = md()
+        super().__init__(
+            name, metadata, query_template, inputs, outputs=outputs, task_config=task_config, *args, **kwargs
```
Comment (katrogan) [inline flytekit/taskplugins/sqlite/task.py:88]: is task_type unchanged from the parent class?

File: tests/flytekit/unit/taskplugins/sqlite/test_task.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    assert df is not None
+
+
+def test_task_dynamic():
+    tk = SQLite3Task("test", query_template="select * from tracks", task_config=SQLite3Config(compressed=True))
+    with pytest.raises(AssertionError):
+        tk()
+
+    df = tk(sqlite=FlyteFile.from_path(EXAMPLE_DB))
+    print(df)
```
Comment (katrogan) [inline tests/flytekit/unit/taskplugins/sqlite/test_task.py:30]: remove?

**Review by katrogan — APPROVED** [review]

File: flytekit/taskplugins/sqlite/task.py (hunk 1 above)
Comment (kumare3) [reply flytekit/taskplugins/sqlite/task.py:34]: +1

File: flytekit/taskplugins/sqlite/task.py (hunk 2 above)
Comment (kumare3) [reply flytekit/taskplugins/sqlite/task.py:88]: good catch, it does not really matter, as this is a python task, but has a lot of benefits for the UI

File: tests/flytekit/unit/taskplugins/sqlite/test_task.py (hunk 3 above)
Comment (kumare3) [reply tests/flytekit/unit/taskplugins/sqlite/test_task.py:30]: sure

File: flytekit/taskplugins/sqlite/task.py (hunk 2 above)
Comment (kumare3) [reply flytekit/taskplugins/sqlite/task.py:88]: i actually fixed it in my next PR, if you guys are ok, I will merge that PR into this one?

File: flytekit/taskplugins/sqlite/task.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+    ):
+        if not inputs:
+            inputs = {}
+        if task_config is not None and task_config.sqlite_db_mode == SQLite3Config.Mode.STATIC:
+            if task_config.static_file_uri is None:
+                raise ValueError("SQLite DB mode configured to be static, but, the static_file_path was not specified.")
+        else:
+            if self._SQLITE_INPUT_FILE in inputs:
+                raise ValueError(f"Input name `{self._SQLITE_INPUT_FILE}` is reserved as an auto-injected input")
+            inputs[self._SQLITE_INPUT_FILE] = FlyteFile
```
Comment (wild-endeavor) [inline flytekit/taskplugins/sqlite/task.py:81]: no format?  doesn't match the comment.

File: flytekit/types/flyte_file.py [hunk 5]
```diff
@@ -15,6 +16,20 @@ def noop():
     ...
 
 
+def unarchive_file(local_path: str, to_dir: str):
```
Comment (wild-endeavor) [inline flytekit/types/flyte_file.py:19]: don't feel like this belongs in this file.

File: tests/flytekit/unit/taskplugins/sqlite/test_task.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        task_config=SQLite3Config(
+            sqlite_db_mode=SQLite3Config.Mode.STATIC, static_file_uri=EXAMPLE_DB, compressed=True,
+        ),
+    )
+    df = tk()
+    assert df is not None
+
+
+def test_task_dynamic():
+    tk = SQLite3Task("test", query_template="select * from tracks", task_config=SQLite3Config(compressed=True))
```
Comment (wild-endeavor) [inline tests/flytekit/unit/taskplugins/sqlite/test_task.py:26]: Can we add an example with `Resources`?

File: flytekit/taskplugins/sqlite/task.py [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+@dataclass
+class SQLite3Config(object):
+    """
+    Use this configuration to configure if sqlite3 files are passed in dynamically into the task or
+    to configure a path to a static sqlite3 compatible database file that is available either
+      - within the container
+      - or from a publicly downloadable source
+
+    If the Mode is marked dynamic, then a variable called ``sqlite: FlyteFile[".sqlite"]`` will be automatically added
```
Comment (wild-endeavor) [inline flytekit/taskplugins/sqlite/task.py:24]: I dunno, I never really like it when we did this in Presto. I feel like magically adding inputs is weird.  Can we maybe make this two different tasks?

File: tests/flytekit/unit/taskplugins/sqlite/test_task.py (hunk 6 above)
Comment (kumare3) [reply tests/flytekit/unit/taskplugins/sqlite/test_task.py:26]: aah interesting, will that work?

File: flytekit/taskplugins/sqlite/task.py (hunk 7 above)
Comment (kumare3) [reply flytekit/taskplugins/sqlite/task.py:24]: probably you are right, ya adding a new task should be cheap, will do

File: flytekit/types/flyte_file.py (hunk 5 above)
Comment (kumare3) [reply flytekit/types/flyte_file.py:19]: it does not, but we do not have ioutils at the moment

**Comment by kumare3** [comment]
I am going to abort this change @wild-endeavor -This will not work. We need one more thing in the base, I will just merge the metadata change and then work on this independently

**Pull request closed without merging**